opensearchpy/connection/krb_requests.py

```diff
@@ -0,0 +1,64 @@
+"""
+Class for the kerberos authentication.
+During the connection phase, one header is added with the kerberos context
+"""
+import warnings
+import kerberos
+
+from . import RequestsHttpConnection
+
+class RequestsKrbConnection(RequestsHttpConnection):
+    """ Class to connect with Kerberos """
+    def __init__(self,
+                 host="localhost",
+                 port=None,
+                 http_auth=None,
+                 use_ssl=False,
+                 verify_certs=True,
+                 ssl_show_warn=True,
+                 ca_certs=None,
+                 client_cert=None,
+                 client_key=None,
+                 headers=None,
+                 http_compress=None,
+                 cloud_id=None,
+                 api_key=None,
+                 opaque_id=None,
+                 **kwargs
+                 ): # pylint: disable=too-many-locals,too-many-arguments
+        """ Init class """
+        super(RequestsKrbConnection, self).__init__(
+            host=host,
+            port=port,
+            use_ssl=use_ssl,
+            headers=headers,
+            http_compress=http_compress,
+            cloud_id=cloud_id,
+            api_key=api_key,
+            opaque_id=opaque_id,
+            **kwargs
+        )
+        self._set_kerberos_headers(host)
+
+
+    def _set_kerberos_headers(self, host):
+        """do the kerberos magic"""
+        principal = 'HTTP@' + host
+        try:
+            dummy, _krb_context = kerberos.authGSSClientInit(principal)
+        except Exception as message: # pylint: disable=broad-except
```

Comment:
1. Why do we swallow exceptions here to raise something more generic?
2. Is there a way to preserve the inner exception in the re-raised exception?